central/config/datastore/singleton.go

```diff
@@ -79,6 +85,16 @@ func initialize() {
 			PublicConfig:  config.GetPublicConfig(),
 			PrivateConfig: &defaultPrivateConfig,
 		}))
+		return
+	}
+
+	if config.GetPrivateConfig().GetDecommissionedClusterRetention() == nil {
```

Comment:
@clickboo Should this change go to migrations, or is it fine here?

Reply:
I mean, should this population logic for retention config be added to migrator/migrations somewhere instead of here? 

New installs would be taken care of by the block above this.

Reply:
There should be a migration for existing clusters with the defaults added. And for new installs, like you mention, this block should remain.

Reply:
Thanks, I will add that in a separate task and get this in so that UI development can move forward.

Reply:
I don't think migration is needed. In this init function you are pulling out the config from the DB and writing a non-empty value if not set. We always expect a non-zero value for retention. Therefore, once you have written it once, this condition will not be true again. 

I recommend refactoring the code to not write twice `ln84` and `ln94`.

Reply:
Done

Reply:
```suggestion
		privateConfig := config.GetPrivateConfig()
		if privateConfig == nil {
			privateConfig = &defaultPrivateConfig
		} else if config.GetPrivateConfig().GetDecommissionedClusterRetention() == nil {
		        privateConfig.DecommissionedClusterRetention = defaultClusterRetentionConfig.GetDecommissionedClusterRetention()
		}
		
```